pkg/controllers/edgeconnect/deployment/deployment.go

```diff
@@ -156,21 +116,55 @@ func edgeConnectContainer(instance *edgeconnectv1alpha1.EdgeConnect, resource st
 			RunAsUser:                address.Of(unprivilegedUser),
 			RunAsNonRoot:             address.Of(true),
 		},
-		VolumeMounts: []corev1.VolumeMount{
-			{MountPath: consts.EdgeConnectMountPath, Name: consts.EdgeConnectVolumeMountName, ReadOnly: true},
-		},
+		VolumeMounts: prepareVolumeMounts(instance),
 	}
 }
 
-func prepareVolume(clientSecretName string) corev1.Volume {
+func prepareVolumes(instance *edgeconnectv1alpha1.EdgeConnect) []corev1.Volume {
+	volumes := []corev1.Volume{prepareConfigVolume()}
+
+	if instance.Spec.CaCertsRef != "" {
+		volumes = append(volumes, corev1.Volume{
+			Name: consts.EdgeConnectCustomCAVolumeName,
+			VolumeSource: corev1.VolumeSource{
+				ConfigMap: &corev1.ConfigMapVolumeSource{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: instance.Spec.CaCertsRef,
+					},
+					Items: []corev1.KeyToPath{
+						{
+							Key:  consts.EdgeConnectCAConfigMapKey,
+							Path: consts.EdgeConnectCustomCertificateName,
+						},
+					},
+				},
+			},
+		})
+	}
+
+	return volumes
+}
+
+func prepareVolumeMounts(instance *edgeconnectv1alpha1.EdgeConnect) []corev1.VolumeMount {
+	volumeMounts := []corev1.VolumeMount{
+		{MountPath: consts.EdgeConnectConfigPath, Name: consts.EdgeConnectConfigVolumeMountName},
```

Comment:
follow up from my previous review:
this is wrong, this creates a folder called `edgeConnect.yaml`

I used `kubectl debug` to check how it shows up
```
➜  dynatrace-operator git:(configure-edgeconnect-via-edgeconnect-yaml) k debug -n dynatrace provision-test-57585f867f-tx8x9 -it --copy-to=my-debugger --set-image=edge-connect=busybox --container=edge-connect -- sh
If you don't see a command prompt, try pressing enter
~ $ cat edgeConnect.yaml/
cat: read error: Is a directory
~ $ ls edgeConnect.yaml/
edgeConnect.yaml
~ $ cat edgeConnect.yaml/edgeConnect.yaml
name: provision-test
api_endpoint_host: ...
oauth:
    endpoint: ...
    client_id: ...
    client_secret: ...
    resource: ...
```

Reply:
@0sewa0 thanks for troubleshooting, ill fix it.

Reply:
fixed in d5bba271